p2p/pb/proto_services.go

```diff
@@ -184,29 +183,30 @@ func DecodeQuaiResponse(respMsg *QuaiResponseMessage) (uint32, interface{}, erro
 // Converts a custom go type to a proto type and marhsals it into a protobuf message
 func ConvertAndMarshal(data interface{}) ([]byte, error) {
 	switch data := data.(type) {
-	case *types.WorkObject:
-		log.Global.Tracef("marshalling block: %+v", data)
-		protoBlock, err := data.ProtoEncode(types.BlockObject)
-		if err != nil {
-			return nil, err
+	case *types.WorkObject, *types.WorkObjectHeaderView, *types.WorkObjectBlockView:
```

Comment:
we should remove the work object in this case statement then as it should not be used anymore

Reply:
Same as above, if it doesn't break anything I will remove it.